src/node.cc

```diff
@@ -2554,29 +2558,23 @@ extern "C" void node_module_register(void* m) {
   }
 }
 
-struct node_module* get_builtin_module(const char* name) {
-  struct node_module* mp;
+#define GET_MODULE_FN(modlist, flag)                                  \
+  struct node_module* get_ ## modlist ## _module(const char* name) {  \
+    struct node_module* mp;                                           \
+                                                                      \
+    for (mp = modlist_ ## modlist; mp != nullptr; mp = mp->nm_link) { \
+      if (strcmp(mp->nm_modname, name) == 0)                          \
+        break;                                                        \
+    }                                                                 \
+                                                                      \
+    CHECK(mp == nullptr || (mp->nm_flags & flag) != 0);               \
+    return (mp);                                                      \
+  }                                                                   \
 
-  for (mp = modlist_builtin; mp != nullptr; mp = mp->nm_link) {
-    if (strcmp(mp->nm_modname, name) == 0)
-      break;
-  }
-
-  CHECK(mp == nullptr || (mp->nm_flags & NM_F_BUILTIN) != 0);
-  return (mp);
-}
-
-struct node_module* get_linked_module(const char* name) {
-  struct node_module* mp;
-
-  for (mp = modlist_linked; mp != nullptr; mp = mp->nm_link) {
-    if (strcmp(mp->nm_modname, name) == 0)
-      break;
-  }
-
-  CHECK(mp == nullptr || (mp->nm_flags & NM_F_LINKED) != 0);
-  return mp;
-}
+GET_MODULE_FN(builtin, NM_F_BUILTIN);
+GET_MODULE_FN(internal, NM_F_INTERNAL);
+GET_MODULE_FN(linked, NM_F_LINKED);
+#undef GET_MODULE_FN
```

Comment:
Better than what was here before but it would be slightly nicer still as a function that takes the list and the flag as arguments:
```c++
inline node_module* FindModule(node_module* list, const char* name, int flag) {
  // ...
}

node_module* get_builtin_module(const char* name) {
  return FindModule(modlist_builtin, name, NM_F_BUILTIN);
}

// etc.
```
Aside: `get_builtin_module()` and `get_linked_module()` are declared in node.h for (AFAICT) no discernible reason.  They aren't marked `NODE_EXTERN` either so there is no chance they work on Windows.  We should probably just remove them.

Reply:
changed to inline

removed questionable signatures from node.h